### ISSUE TITLE
no-unreachable: break after continue in switch statement incorrectly flagged

### ISSUE THREAD

**Issue by AprilArcus** [eslint/eslint#5123]
```JavaScript
/* eslint no-unreachable: 2 */
var node = document.body.childNodes[0], visibleNodes = [], miscNodes = []
while (node = node.nextSibling) {
  switch (node.nodeType) {
    case Node.ELEMENT_NODE:
      nodes.push(node);
      break;
    case Node.TEXT_NODE:
      nodes.push(node);
      break;
    case Node.COMMENT_NODE:
      continue; // refers to the while() loop, will not terminate execution in this case
      break; // triggers no-unreachable
    default:
      miscNodes.push(node);
  }
}
```

Reported in 2.0.0-beta.3

Seems to be a regression from https://github.com/eslint/eslint/issues/1220 / https://github.com/eslint/eslint/pull/1221

**Comment by eslintbot** [comment]
@AprilArcus Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Sorry, I'm not sure I understand the question correctly. `continue` statement means stop execution and go to the next loop cycle immediately. Which means `break` statement will never be executed and is not needed. Hence the "unreachable" message.

**Comment by AprilArcus** [comment]
Oh excuse me, you're entirely correct. I'll close this issue.

**Comment by gheljenor** [comment]
May i reopen this?

In this case `break;` is not really needed, but... Imagine such case of refactoring:

```(javascript)
switch(a) {
	case 1:
		return "one";
	case 2: 
		return "two";
}
```
Then we decide, to decorate output and do something like this:
```(javascript)
var result;
switch(a) {
	case 1:
		result = "one";
	case 2:
		result = "two";
}
return {value: result};
```

If there is no rule 'no-fallthrough'  - we'll break our code. It seems, better to keep `break;`  statements in switch-case, even they are unreachable.

**Comment by not-an-aardvark** [comment]
@gheljenor To avoid that issue, you should enable the [`no-fallthrough`](http://eslint.org/docs/rules/no-fallthrough) rule.
